src/routers/OrganizationRouter.ts

```diff
@@ -54,13 +54,22 @@ const getAdminRouter = (basePath: string = '/', routePath: string = '/management
         size: Math.min(Math.max(parseInt(<string>pageSize), 0), 25),
       };
 
-      const groups = (await authzService.getGroups()).groups
-        .filter((group) => 'nested' in group)
-        .map(({ name, description, _id: id }) => ({
-          name,
-          description,
-          id,
-        }));
+      const allGroups = (await authzService.getGroups()).groups;
+
+      const groups = [];
+
+      for (const group of allGroups) {
+        if (!('nested' in group)) {
+          continue;
+        }
+
+        groups.push({
+          id: group._id,
+          name: group.name,
+          description: group.description,
+          owners: (await authzService.getGroupOwners(group._id)).map((owner) => owner.email),
+        });
+      }
```

Comment:
This can be optimized a bit, the above checks each group from `allGroups` but one at a time.

Instead, we can probably use the helper function `forEachAsync` from `helpers.util`
```
const nested = groups.filter((g) => 'nested' in group);

const groups = await forEachAsync(nested, async (group) => {
    const owners = await authzService.getGroupOwners(group._id);
    return {
        id: group._id,
        name: group.name,
        description: group.description,
        owners: owners.map((owner) => owner.email)
    }
})
```

The above now does a Promise.all on all of the async tasks (i.e service calls)